File: string divider.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {

	/* Enter your code here. Read input from STDIN. Print output to STDOUT */
//char test[]={'h',' ','i','\n','\0'};
    //printf("%s",test);
    int T;
    int outStrIndx=0;
    scanf("%d\n",&T);
    if (0<T<11){
        char inS[10000];
        char outS[100030];
        for (int i;i<T;i++){
            //fgets(inS,10000,stdin);
            scanf("%s\n",inS);
            int inStrL=strlen(inS);
            if (2<=inStrL<=10000){
                //puts(inS);
                for (int inStrIndx=0;inStrIndx<inStrL;inStrIndx+=2){
                    if (inS[inStrIndx]!='\0'){
                        outS[outStrIndx]=inS[inStrIndx];
                        //printf("inStrIndx=%d\n",inStrIndx);
                        //printf("outStrIndx=%d\n",outStrIndx);
                        //printf("outS[outStrIndx]=%c\n",outS[outStrIndx++]);
                        outStrIndx++;
                        }
                }
                //outStrIndx++;
                outS[outStrIndx]=' ';
                //printf("outStrIndx=%d\n",outStrIndx);
                //printf("outS[outStrIndx]=%c\n",outS[outStrIndx++]);
                //outStrIndx++;
                //outS[outStrIndx]=' ';
                //printf("outStrIndx=%d\n",outStrIndx);
                //printf("outS[outStrIndx]=%c\n",outS[outStrIndx++]);
                outStrIndx++;
                for (int inStrIndx=1;inStrIndx<inStrL;inStrIndx+=2){
                    if (inS[inStrIndx]!='\0'){
                        outS[outStrIndx]=inS[inStrIndx];
                        //printf("inStrIndx=%d\n",inStrIndx);
                        //printf("outStrIndx=%d\n",outStrIndx);
                        //printf("outS[outStrIndx]=%c\n",outS[outStrIndx++]);
                        outStrIndx++;
                        }
                }
                //outStrIndx++;
                outS[outStrIndx]='\n';
                outStrIndx++;
            }
        }
        puts(outS);
    }
	return 0;
}

File: string divider.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scanner = new Scanner(System.in);
        int T = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        //int outStrIndx=0;
        if ((0<T)&(T<11)){
             //String outS="";
             for (int i=0;i<T;i++){
                 String outS = "";
                 String inStr=scanner.nextLine();
                 int inStrL=inStr.length();
                 char[] inS=inStr.toCharArray();
            
                if ((2<=inStrL)&(inStrL<=10000)){
                    for (int inStrIndx = 0; inStrIndx < inStrL; inStrIndx += 2) {
                        //if (inS[inStrIndx] != '\0') {
                            outS += inS[inStrIndx];
                            //outStrIndx += 1;
                        //}
                    }

                    outS += ' ';

                    for (int inStrIndx = 1; inStrIndx < inStrL; inStrIndx += 2) {
                       // if (inS[inStrIndx] != '\0') {
                            outS += inS[inStrIndx];
                            //outStrIndx += 1;
                        //}
                    }

                    //outS += '\n'; 
                    System.out.println(outS);  
                }   
                

             }
            

            //System.out.println(outS);
        
        }
          scanner.close(); 
    }
}

